mlflow/sagemaker/__init__.py

```diff
@@ -134,8 +139,8 @@ def push_image_to_ecr(image=DEFAULT_IMAGE_NAME):
     os.system(cmd)
 
 
-def deploy(app_name, model_path, execution_role_arn, bucket, run_id=None,
-           image="mlflow_sage", region_name="us-west-2"):
+def deploy(app_name, model_path, bucket=None, image_url=DEFAULT_IMAGE_URL, run_id=None, 
```

Comment:
also this breaks https://github.com/databricks/mlflow/blob/ed52e68be2a8dd9439eaaf4637e86ad3dc9e6fb4/mlflow/sagemaker/cli.py#L27 ? we should test the cli commands with these changes.

Reply:
Agreed. @tomasatdatabricks - what are your thoughts?

Reply:
Hmm I think it's early enough to get away with breaking API changes although I would double check with Matei on that. I agree with @sueann that we should also update the cli. Not sure whether we want to pass the container url or stick to using the name in cli. 

We should also think through what happens if someone wants to built their custom image and use it instead of the default one. E.g. Should build container return url now?

Reply:
1. we should change the argument ordering back to what it was
2. decide whether to change the image name to url or not (what this allows is being able to use another account's ECR repo, which might be useful - e.g. sagemaker has published tensorflow images you can manually use in their account)
3. see if we can infer the region name automatically (at least on databricks) - i don't think you can have the cluster in one region and do all the operation in another?? (to be checked)